msg-socket/src/req/socket.rs

```diff
@@ -61,18 +64,39 @@ where
         response_rx.await.map_err(|_| ReqError::SocketClosed)?
     }
 
-    /// Connects to the target with the default options.
+    /// Connects to the target with the default options. WARN: this will block until the connection can be established.
```

Comment:
nit: it's async so it won't block the thread, but rather awaits

Reply:
Resolved in upstack PR!